backend/src/export_api.rs

```diff
@@ -0,0 +1,54 @@
+//! Database export functionality for admin users
+
+#[cfg(feature = "server")]
+use crate::get_data_download_secret;
+#[cfg(feature = "server")]
+use dioxus::logger::tracing::info;
+use dioxus::prelude::*;
+#[cfg(feature = "server")]
+use pslink_shared::apirequests::users::Role;
+
+/// Generate a database export URL with the secret token for admin users
+///
+/// # Errors
+/// Fails with [`ServerFnError`] if the user is not authenticated as admin.
+#[server(GetExportUrl, endpoint = "get_export_url")]
+pub async fn get_export_url() -> Result<String, ServerFnError> {
```

Comment:
The secret should be something the user must know to download the database, so the get_export_url is not really necessary.